realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -346,6 +349,13 @@ private static void initializeRealm(Realm realm) {
                 RealmSchema schema = new RealmSchema(realmObjectSchemas);
                 // Assumption: when SyncConfiguration then additive schema update mode
                 realm.sharedRealm.updateSchema(schema, version);
+                // FIXME - Begin: OS currently grabs its own write transaction which prevents us from doing it in Java
```

Comment:
Object store's `update_schema()` has a parameter to say that you are already in a transaction.